routers/api/packages/container/container.go

```diff
@@ -148,16 +150,29 @@ func DetermineSupport(ctx *context.Context) {
 // If the current user is anonymous, the ghost user is used unless RequireSignInView is enabled.
 func Authenticate(ctx *context.Context) {
 	u := ctx.Doer
+	packageScope := auth_service.GetAccessScope(ctx.Data)
 	if u == nil {
 		if setting.Service.RequireSignInView {
 			apiUnauthorizedError(ctx)
 			return
 		}
 
 		u = user_model.NewGhostUser()
+	} else {
```

Comment:
If `RequireSignInView` is false, then no-login user can access anything but logged in user with no scope can not access.
Is it correct? 🤔

Reply:
This function `Authenticate` just return the token based on the login information. For anonymous user, the scope will be "", so we don't need to check it. For those login with access token with scope, we have to check the scopes.

The returned token will be taken in the following requests, if scope is empty, there are only public package read cannot be done.